json-schema/json-schema/src/test/scala/endpoints4s/TuplerTests.scala

```diff
@@ -5,19 +5,56 @@ class TuplerTests {
   def tupling[A, B](a: A, b: B)(implicit tupler: Tupler[A, B]): tupler.Out =
     tupler(a, b)
 
-  def forall[A, B, C, D, E, F](a: A, b: B, c: C, d: D, e: E, f: F): Unit = {
+  def forall[A, B, C, D, E, F, G, H, I](a: A, b: B, c: C, d: D, e: E, f: F, g: G, h: H, i: I): Unit = {
+    // Left/Right unit
+    tupling((), ()): Unit
+    tupling((), a): A
+    tupling(a, ()): A
+
+    // Tupler
     tupling(a, b): (A, B)
     tupling(a, (b, c)): (A, B, C)
-    tupling((a, b), c): (A, B, C)
     tupling((a, b), (c, d)): (A, B, C, D)
     tupling(a, (b, c, d, e)): (A, B, C, D, E)
     tupling((a, b), (c, d, e)): (A, B, C, D, E)
+
+    // TuplerAppend
+    tupling((a, b), c): (A, B, C)
     tupling((a, b, c), d): (A, B, C, D)
     tupling((a, b, c, d), e): (A, B, C, D, E)
     tupling((a, b, c, d, e), f): (A, B, C, D, E, F)
-    tupling(a, ()): A
-    tupling((), a): A
-    tupling((), ()): Unit
+    tupling((a, b, c, d, e, f), g): (A, B, C, D, E, F, G)
+    tupling((a, b, c, d, e, f, g), h): (A, B, C, D, E, F, G, H)
+    tupling((a, b, c, d, e, f, g, h), i): (A, B, C, D, E, F, G, H, I)
+
+    // Stability tests
```

Comment:
Not sure what you mean by “stability” test, do you mind elaborating?

Reply:
"Stability of implementation", whether a proposed change alters the way things stand today. Any alterations to these "out of spec" tests would be reflected in a change to these lines, signaling to both maintainers and users of this library when the semantics change.